patches/node/feat_expose_report_writing_entrypoint_to_embedders.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: deepak1556 <hop2deep@gmail.com>
+Date: Wed, 18 Sep 2024 22:27:07 +0900
+Subject: feat: expose report writing entrypoint to embedders
+
+Allows creating Node.js diagnostic reports from Electron signal handlers.
+Currently used by Utility process to support error event.
+
+diff --git a/src/node_report.cc b/src/node_report.cc
+index 5368d8eef2fac006bd855dc5e0e1545c90a18cc8..80e3f93a4d96d3f23622eaccc5955f7d63d2e466 100644
+--- a/src/node_report.cc
++++ b/src/node_report.cc
+@@ -54,15 +54,6 @@ using v8::Value;
+ 
+ namespace report {
+ // Internal/static function declarations
+-static void WriteNodeReport(Isolate* isolate,
+-                            Environment* env,
+-                            const char* message,
+-                            const char* trigger,
+-                            const std::string& filename,
+-                            std::ostream& out,
+-                            Local<Value> error,
+-                            bool compact,
+-                            bool exclude_network = false);
+ static void PrintVersionInformation(JSONWriter* writer,
+                                     bool exclude_network = false);
+ static void PrintJavaScriptErrorStack(JSONWriter* writer,
+@@ -88,15 +79,15 @@ static void PrintNetworkInterfaceInfo(JSONWriter* writer);
+ 
+ // Internal function to coordinate and write the various
+ // sections of the report to the supplied stream
+-static void WriteNodeReport(Isolate* isolate,
+-                            Environment* env,
+-                            const char* message,
+-                            const char* trigger,
+-                            const std::string& filename,
+-                            std::ostream& out,
+-                            Local<Value> error,
+-                            bool compact,
+-                            bool exclude_network) {
++void WriteNodeReport(Isolate* isolate,
```

Comment:
@deepak1556 have you attempted to upstream this to Node.js?

Reply:
Not yet

Reply:
Patch has been removed in favor of the public API